### ISSUE TITLE
Commander Temp sensor accuracy

### ISSUE THREAD

**Issue by kick81** [iJacks1980/FanControl.CommanderPRO#20]
Hi, 

Is it possible to adjust the accuracy of the temp sensors?
I´m trying to make a delta temp curve using 2 different t-probes (ambient and coolant) but the temperatures are not accurate. 
For example temp in idle marks 28 and when the water is warming up 30 and then jumps to 33 and so...

Thanks!!

**Comment by iJacks1980** [comment]
What device are you using, a Commander PRO, CORE, or CORE XT?

**Comment by kick81** [comment]
Is the commander pro

**Comment by iJacks1980** [comment]
Can you please attach the .trc file for the Commander PRO plugin.

The temps that get reported back to Fan Control are what the device sends back - but the PRO temps are only from the two attachable probes...which won't give the coolant temperature.

**Comment by kick81** [comment]
I can´t find any .trc file in my system, sorry I´m totally noob. Hoy can I create it?

Here are the sensor using ICUE

<blockquote class="imgur-embed-pub" lang="en" data-id="a/6O89FDo" data-context="false" ><a href="//imgur.com/a/6O89FDo"></a></blockquote><script async src="//s.imgur.com/min/embed.js" charset="utf-8"></script>

And here are the sensor in Fancontrol

<blockquote class="imgur-embed-pub" lang="en" data-id="a/tAiwQSK" data-context="false" ><a href="//imgur.com/a/tAiwQSK"></a></blockquote><script async src="//s.imgur.com/min/embed.js" charset="utf-8"></script>

You can see that there are 4 t-probes, with ICUE I can see each temperature precisely, updating it with decimals, but with fancontrol and your plugin, temps are refreshing jumping between 20-23-25-28-30-33...
It´s like the plugin deletes steps between ºC´s

Thanks for your effort!

**Comment by iJacks1980** [comment]
If iCUE is installed it won't be able to read the temp (or fan) data correctly.  It's one or the other, sadly due to the way the device protocol works there is no way around this.

The trc file should be located in the fancontrol main folder, called CommanderPRO.trc - depending on the version you are using.

**Comment by iJacks1980** [comment]
This version will deffo produce a trc file: [https://github.com/iJacks1980/FanControl.CommanderPRO/files/8748698/FanControl.CommanderPro.zip](https://github.com/iJacks1980/FanControl.CommanderPRO/files/8748698/FanControl.CommanderPro.zip)

**Comment by kick81** [comment]
Hi, Could you take a look at it?

**Comment by iJacks1980** [comment]
Still not seeing the issue, but you have labelled the temperature probes incorrectly - the PRO does not have a water temp, only the 4 that you plug in.  So no idea how you expect to make some kind of delta between "hot" and "cold" water.  I can only assume you have placed the probes in places like on the pump block and the radiator?

The first releases did have an issue where it only returned whole numbers, but the version I linked above has that fixed.  But there are no missing temperatures.  The plugin reads the temperature when FanControl asks for it to be read - which is every second.  If you want more precision than that you will have to ask Rem0o to support this in FanControl.

Unless I have misunderstood what you mean by "There are missing temps"?

**Comment by kick81** [comment]
Yes, I have 4 sensors connected: 

- Two g1/4 sensors for hot water (before rad) and codl water (after radiators). Like this one https://www.amazon.com/Bitspower-Temperature-Sensor-Fitting-Sparkle/dp/B003U3VTT0
- One T-probe for ambient outside the case
- Another T-probe for ambient inside the case

When I say "missing temps" I mean that there are jumps between reads, for example temp value jumps from 30c to 33c (31c and 32c are missing) and there are no decimals at all.

Maybe is an issue with Fancontrol, I´ll ask to Rem0o.

Thanks for all!

**Comment by iJacks1980** [comment]
Ahh, so I assume you are using the PRO as a glorified fan controller and temp hub rather than controlling the stock water pump?  Or you did the brave thing and replaced the tubing?  Not that it matters, I wasn't aware such a device existed - I have not gone down into custom loop components yet (probably will if/when I upgrade to a Ryzen 7950x).

So now I understand what you mean I don't think my plugin can "fix" this issue, as the iCUE devices are all polled devices they only have the ability to report sensor values when requested, so if a temp does change by 2 degrees it won't see that.  I'll ask Rem0o if it is possible to have a per plugin poll timer.  Mind you, it should be showing temps in base 10...so why your not seeing decimals I am not sure.

What version of FanControl are you using?  And is it the .Net 4.8 or .Net 6 version?  I've not tested with the .Net 6 version yet.

**Comment by kick81** [comment]
Yes!! it´s a custom loop. And 3c/5c degrees in water temp is a lot, and without accurate values my fan curve goes crazy from 400rpm to 1000rpm when the temps jumps.  

I wanted to buy the aquacomputer octo/quadro , but there was no stock and finally bought the commander pro (my motherboard has only one sensor and need at least 2 to make a delta curve)

Is the .Net 4.8 version, I´ve tested the .Net 6 version and there is the same issue.

**Comment by Rem0o** [comment]
@iJacks1980  you can always have your own thread spinning, which poll at the frequency you like, and push the updated value. 

I think though the problem @kick81 mentions is that he expects float values where as from here https://github.com/iJacks1980/FanControl.CommanderPRO/blob/a582528af0a19b2413ecaf4f17e09c2bfeadfee5/FanControl.CommanderPro/Device.cs#L268 the buffer is read as a int16.

**Comment by iJacks1980** [comment]
@Rem0o, the code in here is a little out of date - my local version had it as a decimal, but I had backed that change out by mistake.

@kick81 I'll work on an update tonight to get the decimal back in - erm, I no longer have a Commander PRO connected (mainboard only has a single USB header and I don't have any splitters available).  So there might be some back and forth with test versions.  Especially with setting the refresh rate, but I'll test that part with my Commander CORE first.

**Comment by kick81** [comment]
Hi @iJacks1980 ,

Any news?

**Comment by iJacks1980** [comment]
Hi @kick81,

I'm still working on it, sadly I've been dealing with some serious family issues which have taken most of my time and I'm going to be away for a few more days.  But I'll try and get an interim update ready tonight (might be quite late).

**Comment by kick81** [comment]
Hi @iJacks1980,
I hope everything is better with your family, remember when you have time to upload the update.

Thanks!!!
